src/modules/quick_method_module.f90

```diff
@@ -538,13 +537,17 @@ subroutine read_quick_method(self,keywd)
             ! Density map
             ! JOHN FAVER 12/2008
             if (index(keywd,'DENSITYMAP') /= 0) then
-                self%gridspacing = rdnml(keywd,'DENSITYMAP')
+                i = index(keywd,'DENSITYMAP=',.false.)
+                j = scan(keywd(i:len_trim(keywd)), ' ', .false.)
+                read(keywd(i+10:i+j-2), *) self%gridspacing
```

Comment:
This is not robust and leads to repetitive code. You want to write a read function only once and then use it multiple times so that you will have code like

```
user input_parser, only :: read
...
if (index(input_string,'FOO') /= 0) then
   self%gridspacing = read(input_string, 'FOO')
end if
```

Then you write a module with robust functions for different data types and use interface to overload the read function.

```
module input_parser

  implicit none

  private
  public :: read

  interface read
    module procedure read_integer_keyword
    module procedure read_float_keyword
    module procedure read_string_keyword
  end interface

contains

  function read_integer_keyword(input, keyword)
    implicit none
    character(len=*), intent(in) :: input
    character(len=*), intent(in) :: keyword
    integer :: read_integer_keyword

    FOLLOWS YOUR ROBUST IMPLEMENTATION TO PARSE THE STRING

     read_integer_string = WHATEVER YOU PARSED

  function read_integer_keyword

  IMPLEMENTATION OF FUNCTIONS FOR OTHER DATA TYPES

end module input_parser
```

Of course it makes sense in the above to have a robust function internal to the parser module (i.e not public) that identifies the piece of the string that contains the data so you don't do this in every of the above functions.